.cbmc-batch/jobs/aws_cryptosdk_sig_get_privkey/aws_cryptosdk_sig_get_privkey_harness.c

```diff
@@ -23,28 +23,28 @@
 #include <cipher_openssl.h>
 
 void aws_cryptosdk_sig_get_privkey_harness() {
-    /* arguments */
+    /* Nondet Input */
     struct aws_cryptosdk_sig_ctx *ctx = can_fail_malloc(sizeof(struct aws_cryptosdk_sig_ctx));
     struct aws_allocator *alloc       = can_fail_allocator();
     struct aws_string *priv_key;
 
-    /* assumptions */
-    __CPROVER_assume(ctx);
+    /* Assumptions */
+    __CPROVER_assume(ctx != NULL);
```

Comment:
What we really want here is
```c
struct aws_cryptosdk_sig_ctx *ctx = ensure_sig_ctx_has_allocated_members();
```
And inside `ensure_sig_ctx_has_allocated_members` we nondeterministically allocate memory or `ctx`. I'd also expect that `aws_cryptosdk_sig_ctx_is_valid_cbmc` ensures `ctx` is different than NULL, which would avoid the need of the extra assumption. Why do we have a `cbmc` sufix in the name of that function?

Reply:
Yep, updated as in https://github.com/aws/aws-encryption-sdk-c/pull/602.